CodeEdit/Documents/WorkspaceDocument.swift

```diff
@@ -340,27 +328,19 @@ import TabBar
 
     // MARK: Close Workspace
 
+    private func saveSelectionState() throws {
+        guard let path = fileURL?.path,
+              let pathData = path.data(using: .utf8) else { return }
+        let hash = String(CryptoKit.SHA256.hash(data: pathData).hashValue)
```

Comment:
`.hashValue` will not be equal across executions of the program ([source](https://developer.apple.com/documentation/swift/string/hashvalue)).

Consider implementing it something like that:

```swift
let stringData = "Hello World".data(using: .utf8)!

let hash = SHA256.hash(data: stringData)

let hashString = hash.compactMap { String(format: "%02x", $0) }.joined()
```